### ISSUE TITLE
Build failures with mtl-2.3

### ISSUE THREAD

**Issue by sjakobi** [phile314/tasty-silver#35]
When I build `tasty-silver-3.2` and up with `mtl-2.3` and the current `tasty` `master`, I'm getting build errors like these:

```
Test/Tasty/Silver/Interactive.hs:258:3: error:
    Variable not in scope: when :: Bool -> IO () -> IO a7
    |
258 |   when isTerm showCursor
    |   ^^^^

Test/Tasty/Silver/Interactive.hs:264:9: error:
    Variable not in scope: when :: Bool -> IO () -> IO a8
    |
264 |         when isTerm hideCursor
    |         ^^^^

Test/Tasty/Silver/Interactive.hs:269:9: error:
    Variable not in scope: when :: Bool -> IO () -> IO a9
    |
269 |         when isTerm hideCursor
    |         ^^^^

Test/Tasty/Silver/Interactive.hs:334:11: error:
    Variable not in scope: when :: Bool -> IO () -> IO a10
    |
334 |           when (resultTime result >= 0.01 || forceTime) $
    |           ^^^^

Test/Tasty/Silver/Interactive.hs:347:11: error:
    Variable not in scope: when :: Bool -> IO () -> IO a11
    |
347 |           when (not $ null rDesc) $ (case getResultType result of
    |           ^^^^

Test/Tasty/Silver/Interactive.hs:410:11: error:
    Variable not in scope: when :: Bool -> IO () -> IO a12
    |
410 |           when (not $ null rDesc) $ (case getResultType result'' of
    |           ^^^^

Test/Tasty/Silver/Interactive.hs:506:9: error:
    Variable not in scope: when :: Bool -> t -> f a3
    |
506 |         when nonempty $ printHeading
    |         ^^^^

Test/Tasty/Silver/Interactive.hs:531:9: error:
    Variable not in scope: unless :: Bool -> IO () -> f a4
    |
531 |         unless failed clearAboveLine
    |         ^^^^^^

Test/Tasty/Silver/Interactive.hs:562:9: error:
    Variable not in scope: unless :: Bool -> m0 () -> f a5
    |
562 |         unless failed $
    |         ^^^^^^

Test/Tasty/Silver/Interactive.hs:598:3: error:
    Variable not in scope: when :: Bool -> t0 -> IO a0
    |
598 |   when (statCreatedGolden st > 0) (printf "Created %d golden values.\n" (statCreatedGolden st))
    |   ^^^^

Test/Tasty/Silver/Interactive.hs:599:3: error:
    Variable not in scope: when :: Bool -> t1 -> IO a1
    |
599 |   when (statUpdatedGolden st > 0) (printf "Updated %d golden values.\n" (statUpdatedGolden st))
    |   ^^^^

Test/Tasty/Silver/Interactive.hs:600:3: error:
    Variable not in scope: when :: Bool -> t2 -> IO a2
    |
600 |   when (statDisabled st > 0) (printf "Ignored %d disabled tests.\n" (statDisabled st))
    |   ^^^^
cabal: Failed to build tasty-silver-3.2
```

As a Hackage trustee I have created revisions for these versions that should prevent users from encountering these errors. See e.g. https://hackage.haskell.org/package/tasty-silver-3.3.1/revisions/

**Comment by andreasabel** [comment]
I can confirm this with the following `cabal.project.local`:
```
constraints: mtl == 2.3
allow-newer: mtl

-- tasty
------------------------------------------------------------------------

source-repository-package
  type: git
  location: https://github.com/UnkindPartition/tasty
  subdir: core

-- random
------------------------------------------------------------------------

source-repository-package
  type: git
  location: https://github.com/haskell/random.git
  tag: 5ecc8935994159c6d39088233a0887714aacaf5e

constraints: random == 1.3.*
allow-newer: *:random
```

**Comment by andreasabel** [comment]
Published v3.3.1.1: https://hackage.haskell.org/package/tasty-silver-3.3.1.1

**Comment by sjakobi** [comment]
Thank you, @andreasabel! :)
